deluge/common.py

```diff
@@ -938,12 +938,13 @@ def is_ipv4(ip):
     import socket
 
     try:
-        if windows_check():
-            return socket.inet_aton(ip)
-        else:
-            return socket.inet_pton(socket.AF_INET, ip)
+        for interface in ip.split(','):
```

Comment:
why split the `ip`? are we expecting it to have several options?
the same goes for `is_ipv6` function

Reply:
Yes for multi-homing.

Reply:
We could do the for loop else where where we need to verify interfaces/ips I guess.

Reply:
I asked because running it on the entire string will fail in case it contain several types of interfaces:
```
>>> is_ipv4('127.0.0.1,eth01')
Unable to verify IPv6 Address with socket.
False
```

Reply:
True. Would it work to extend is_ip to return return is_ipv4(ip) or is_ipv6(ip) or is_net_interface(ip)

Reply:
No that would still fail. Hmmm not sure how to fix this.

Reply:
I think I need to move the for loop out of the ip_* etc checks.

Reply:
yep. maybe the outer function I suggested in the opening message.
For example (the name needs to be changed 😅):
```python
def verify_ip_or_interface(ip):
    interfaces = []
    for interface in ip.split(','):
        if is_ip(interface) or is_net_interface(interface):
            interfaces.apped(interface)
    return ','.join(interfaces)
```

Reply:
Honestly I think Multi-Homing is above my understanding.

Reply:
don't worry, eventually we will get it also :)

Reply:
I added a test but still not sure how to clean up if is_ip(inter) or is_net_interface(inter)

Reply:
@DjLegolas would something like https://github.com/deluge-torrent/deluge/pull/338/commits/7384191554606094c94813effce5270e314d7219 work?